artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -198,6 +203,18 @@ public AMQPConnectionContext(ProtonProtocolManager protocolManager,
       }
    }
 
+   public List<AMQPMirrorControllerTarget> getMirrorControllerTargets() {
+      return mirrorControllerTargets;
+   }
+
+   public AMQPConnectionContext addMirrorControllerTarget(AMQPMirrorControllerTarget mirrorControllerTarget) {
+      if (mirrorControllerTargets == null) {
+         mirrorControllerTargets = new ArrayList<>();
+      }
+      mirrorControllerTargets.add(mirrorControllerTarget);
```

Comment:
There is no remove anywhere, so this is already a potential leak.

Reply:
not really.. the connection will be removed.... the connection will be gone in case of a disconnect.